### PR TITLE
Add Helm chart with unit tests and Makefile targets

### LINKED ISSUE #143: PR 6/7: Helm chart + Dockerfile + Makefile targets
Parent: #137
Depends on: #142

## Scope

Deployment infrastructure: Dockerfile, Helm chart from scratch, Makefile targets.

**Branch:** `pr/06-helm-dockerfile` from `pr/05-controller-tests`
**Targets:** `pr/05-controller-tests`

## Dockerfile

Multi-stage build:
```dockerfile
FROM golang:1.25 AS builder
WORKDIR /workspace
COPY go.mod go.sum ./
RUN go mod download
COPY . .
RUN CGO_ENABLED=0 GOOS=linux go build -o manager cmd/main.go

FROM gcr.io/distroless/static:nonroot
WORKDIR /
COPY --from=builder /workspace/manager .
USER 65532:65532
ENTRYPOINT ["/manager"]
```

Pure-Go ISO extraction — no external tools, distroless works.

## Helm chart: `charts/isoboot/`

```
charts/isoboot/
  Chart.yaml
  values.yaml
  crds/
    bootsource.yaml      # synced from controller-gen output
  templates/
    _helpers.tpl         # standard name/labels/selectors helpers
    deployment.yaml      # controller-manager
    serviceaccount.yaml
    rbac.yaml            # ClusterRole + ClusterRoleBinding
```

### `values.yaml`

```yaml
image:
  repository: ghcr.io/isoboot/isoboot
  tag: ""  # defaults to .Chart.AppVersion
  pullPolicy: IfNotPresent

controller:
  baseDir: /data/bootsource
  resources:
    limits:
      memory: 512Mi
      cpu: 500m
    requests:
      memory: 256Mi
      cpu: 250m

serviceAccount:
  create: true
  name: ""
```

### `deployment.yaml` key points

- Single container: `/manager --base-dir={{ .Values.controller.baseDir }}`
- Health probes: `:8081/healthz` and `:8081/readyz` (controller-runtime defaults)
- Volume mount for boot files (hostPath or PVC, configurable)

### `rbac.yaml`

- ClusterRole: get/list/watch/update/patch on `bootsources` and `bootsources/status`
- ClusterRoleBinding → ServiceAccount

## Makefile additions

Add to kubebuilder-generated Makefile:
- `helm-sync`: `cp config/crd/bases/isoboot.github.io_bootsources.yaml charts/isoboot/crds/bootsource.yaml`
- `helm-lint`: `helm lint charts/isoboot/`
- `helm-template`: `helm template test charts/isoboot/`

## Verify

```bash
make helm-sync
make helm-lint
helm template test charts/isoboot/
docker build -t ghcr.io/isoboot/isoboot:dev .
```

### PR DESCRIPTION
**Pull request by twdamhore**
## Summary

- Add complete Helm chart (`charts/isoboot/`) for deploying the isoboot controller
- Add JSON Schema validation (`values.schema.json`) for chart values
- Add 60 helm-unittest tests covering all templates with positive and negative cases
- Add Makefile targets: `helm-sync`, `helm-lint`, `helm-template`, `helm-test`

Closes #143

### Helm Chart Structure
```
charts/isoboot/
├── Chart.yaml           # Chart metadata (v0.1.0)
├── values.yaml          # Default configuration
├── values.schema.json   # JSON Schema validation
├── .helmignore          # Package exclusions
├── crds/
│   └── bootsource.yaml  # Synced from controller-gen
└── templates/
    ├── _helpers.tpl     # Name/label/selector helpers
    ├── deployment.yaml  # Controller deployment (with optional init container)
    ├── serviceaccount.yaml
    ├── rbac.yaml        # ClusterRole + ClusterRoleBinding + Role + RoleBinding
    └── pvc.yaml         # Optional PVC for storage
```

### Key Features
- **Storage options**: hostPath (default) or PVC
- **Security**: Non-root user 65532, read-only rootfs, dropped capabilities, fsGroup for PVC
- **Health probes**: /healthz and /readyz on :8081
- **baseDir**: `/var/lib/isoboot` (matches cmd/main.go default)
- **Leader election**: Namespaced Role/RoleBinding for coordination.k8s.io/leases
- **Permission handling**: Optional init container for hostPath, fsGroup for PVC

### RBAC Resources
| Resource | Scope | Purpose |
|----------|-------|---------|
| ClusterRole | Cluster | bootsources CRUD, events |
| ClusterRoleBinding | Cluster | Binds ClusterRole to ServiceAccount |
| Role | Namespace | coordination.k8s.io/leases for leader election |
| RoleBinding | Namespace | Binds Role to ServiceAccount |

### Makefile Targets
| Target | Description |
|--------|-------------|
| `helm-sync` | Sync CRD to chart's crds directory |
| `helm-lint` | Lint the Helm chart |
| `helm-template` | Render templates locally |
| `helm-test` | Run helm-unittest (60 tests) |

## Test Plan

- [x] `make helm-lint` passes (1 chart linted, 0 failed)
- [x] `make helm-template` renders valid YAML
- [x] `make helm-test` passes (60/60 tests)
- [x] `make build` passes
- [x] `make lint` passes (0 issues)
- [x] `make test` passes (all Go tests)
- [x] `docker build` succeeds

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -248,3 +274,41 @@ endef
...
+  echo "Actual: $$actual"; \
+  rm -rf "$$tmpdir"; \
+  exit 1; \
+fi; \
+echo "Checksum verified"; \
+tar xz -C "$$tmpdir" -f "$$tmpdir/$$tarball"; \
+mv "$$tmpdir/untt" "$(1)-$(2)"; \
+rm -rf "$$tmpdir"; \
+}; \
+ln -sf "$(1)-$(2)" "$(1)"
```
[the remark this answers is not in the document]
Comment (twdamhore) [reply Makefile]: From Claude:

Addressed the supply chain concern by embedding immutable SHA256 checksums directly in the Makefile for all supported platforms (linux/macos, amd64/arm64). The checksums are checked into the repo rather than fetched from the release, providing a stronger security guarantee.

File: charts/isoboot/values.schema.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+          "type": "string",
+          "enum": ["hostPath", "pvc"],
+          "description": "Storage type for boot files"
+        },
+        "hostPath": {
+          "type": "object",
+          "properties": {
+            "path": {
+              "type": "string",
+              "pattern": "^/"
```
[the remark this answers is not in the document]
Comment (twdamhore) [reply charts/isoboot/values.schema.json]: From Claude:

Updated schema to allow empty `storage.hostPath.path` (removed the `pattern` constraint). The deployment template's `default` fallback to `controller.baseDir` now works correctly. Also updated values.yaml to default to empty string with documentation.
